### ISSUE TITLE
leaps: websocket behind reverse-proxy broken

### ISSUE THREAD

**Issue by qknight** [nixcloud/nixcloud-webservices#26]
issue description:

leaps uses a websocket for the document editing. because the Host header is not set, leaps will return a CORS error which can be fixed by adding: `proxy_set_header Host foo.de;` into the reverse-proxy configuration.

the problem is that:

proxyOptions.http. | proxyOptions.https.
* flags
* record

would need to compute `config.proxyOptions.domain` which is not possible at the moment.

maybe it would make sense to have `extraFlags` or also make `flags` and `record` a function which is evaluated before the config is generated inside options.nix from the reverse-proxy. 

```
user "reverse-proxy" "reverse-proxy";
error_log stderr;
daemon off;
events {}
http {
        server {
                listen 80;
                listen [::]:80;
                server_name foo.de;
                location /myleaps {
                        rewrite     ^   https://$server_name$request_uri? permanent;
                }
        }
        server {
                listen 80;
                listen [::]:80;
                server_name example.org;
                location /wiki {
                        rewrite     ^   https://$server_name$request_uri? permanent;
                }
        } server {
                ssl on;
                listen 443 ssl;
                listen [::]:443 ssl;
                server_name foo.de;
                ssl_certificate /var/lib/nixcloud/TLS/foo.de/selfsigned/fullchain.pem;
                ssl_certificate_key /var/lib/nixcloud/TLS/foo.de/selfsigned/key.pem;
                location /myleaps {
                proxy_pass http://${location.ip}:${toString location.port}${location.path};
                }
                location /myleaps/leaps/ws {
                        # https websocket default flags
                        proxy_http_version 1.1;
                        proxy_set_header Upgrade $http_upgrade;
                        proxy_set_header Connection "upgrade";
                        proxy_set_header Host foo.de;
                        proxy_set_header X-Forwarded-For $remote_addr;
                        proxy_pass http://127.0.0.1:3031/myleaps/leaps/ws;
                }
        }
        server {
                ssl on;
                listen 443 ssl;
                listen [::]:443 ssl;
                server_name example.org;
                ssl_certificate /var/lib/nixcloud/TLS/example.org/selfsigned/fullchain.pem;
                ssl_certificate_key /var/lib/nixcloud/TLS/example.org/selfsigned/key.pem;
                location /wiki {
                        # https default flags
                        proxy_set_header Host $host;
                        proxy_set_header X-Real-IP $remote_addr;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_set_header X-Forwarded-Proto $scheme;
                        proxy_pass http://127.0.0.1:5050/wiki;
                }
        }
}

```

**Comment by qknight** [comment]
good news, we can just use: `proxy_set_header Host $host;` instead and don't need the `location`

**Comment by qknight** [comment]
fixed in be57d526547c66db05595002682525ca62c8f068
